ql/src/java/org/apache/hadoop/hive/ql/ddl/table/tag/create/AlterTableCreateTagAnalyzer.java

```diff
@@ -0,0 +1,93 @@
+import org.apache.hadoop.hive.ql.ddl.DDLUtils;
+import org.apache.hadoop.hive.ql.ddl.DDLWork;
+import org.apache.hadoop.hive.ql.ddl.table.AbstractAlterTableAnalyzer;
+import org.apache.hadoop.hive.ql.ddl.table.AlterTableType;
+import org.apache.hadoop.hive.ql.exec.TaskFactory;
+import org.apache.hadoop.hive.ql.hooks.ReadEntity;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.parse.ASTNode;
+import org.apache.hadoop.hive.ql.parse.AlterTableTagSpec;
+import org.apache.hadoop.hive.ql.parse.HiveParser;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.session.SessionState;
+
+import static org.apache.hadoop.hive.ql.parse.AlterTableTagSpec.AlterTagOperationType.CREATE_TAG;
+
+@DDLSemanticAnalyzerFactory.DDLType(types = HiveParser.TOK_ALTERTABLE_CREATE_TAG)
+public class AlterTableCreateTagAnalyzer extends AbstractAlterTableAnalyzer {
+  public AlterTableCreateTagAnalyzer(QueryState queryState) throws SemanticException {
+    super(queryState);
+  }
+
+  @Override
+  protected void analyzeCommand(TableName tableName, Map<String, String> partitionSpec, ASTNode command)
+      throws SemanticException {
+    Table table = getTable(tableName);
+    validateAlterTableType(table, AlterTableType.CREATE_TAG, false);
+    DDLUtils.validateTableIsIceberg(table);
+    inputs.add(new ReadEntity(table));
+
+    String tagName = command.getChild(0).getText();
+    Long snapshotId = null;
+    Long asOfTime = null;
+    Long maxRefAgeMs = null;
+    for (int i = 1; i < command.getChildCount(); i++) {
+      ASTNode childNode = (ASTNode) command.getChild(i);
+      switch (childNode.getToken().getType()) {
+      case HiveParser.TOK_AS_OF_VERSION:
+        snapshotId = Long.parseLong(childNode.getChild(0).getText());
+        break;
+      case HiveParser.TOK_AS_OF_TIME:
+        ZoneId timeZone = SessionState.get() == null ? new HiveConf().getLocalTimeZone() :
+            SessionState.get().getConf().getLocalTimeZone();
+        TimestampTZ ts = TimestampTZUtil.parse(stripQuotes(childNode.getChild(0).getText()), timeZone);
+        asOfTime = ts.toEpochMilli();
+        break;
+      case HiveParser.TOK_RETAIN:
+        String maxRefAge = childNode.getChild(0).getText();
+        String timeUnitOfBranchRetain = childNode.getChild(1).getText();
+        maxRefAgeMs =
+            TimeUnit.valueOf(timeUnitOfBranchRetain.toUpperCase(Locale.ENGLISH)).toMillis(Long.parseLong(maxRefAge));
+        break;
+      default:
+        throw new SemanticException("Unrecognized token in ALTER CREATE TAG statement");
+      }
+    }
+
+    AlterTableTagSpec.CreateTagSpec createTagspec =
+        new AlterTableTagSpec.CreateTagSpec(tagName, snapshotId, asOfTime, maxRefAgeMs);
+    AlterTableTagSpec alterTableTagSpec = new AlterTableTagSpec(CREATE_TAG, createTagspec);
+    AlterTableCreateTagDesc desc = new AlterTableCreateTagDesc(tableName, alterTableTagSpec);
```

Comment:
could we extend from AlterTableCreateBranchAnalyzer and override just AlterTableCreateTagDesc creation (i.e. getAlterTableDesc or something)?